src/main/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+        CheckReconciliation checkReconciliation = checkReconciliationDao.findByCheckNumber(staleCheckRow.getCheckNumber());
+
+        String processingError = validateStaleCheckBatchRow(staleCheckRow, checkReconciliation);
+        if (StringUtils.isNotBlank(processingError)) {
+            return processingError;
+        }
+        checkReconciliation.setStatus(CRConstants.STALE);
+        checkReconciliation.setStatusChangeDate(dateTimeService.getCurrentSqlDate());
+        businessObjectService.save(checkReconciliation);
+
+        return processingError;
+    }
+
+    protected String validateStaleCheckBatchRow(StaleCheckBatchRow staleCheckRow, CheckReconciliation checkReconciliation) {
+        List<String> validationErrors = new ArrayList<>();
+
+        if (StringUtils.isBlank(staleCheckRow.getBankCode())) {
+            validationErrors.add("Bank is blank.");
+        }
+
+        if (StringUtils.isBlank(staleCheckRow.getCheckNumber())) {
+            validationErrors.add("Check Number is blank.");
+        }
+
+        if (StringUtils.isBlank(staleCheckRow.getCheckTotalAmount())) {
+            validationErrors.add("Check Total Amount is blank.");
+        }
+
+        if (ObjectUtils.isNull(checkReconciliation) || ObjectUtils.isNull(checkReconciliation.getCheckNumber())) {
+            validationErrors.add("Check Reconciliation does not exist in KFS.");
+        } else {
+            validateCheckReconciliation(staleCheckRow, checkReconciliation, validationErrors);
+        }
+
+        if (validationErrors.size() > 0) {
+            String validationErrorMessagesForRow = getValidationErrorsForRow(validationErrors);
+            LOG.warn("validateStaleCheckBatchRow:" + validationErrorMessagesForRow);
+            return validationErrorMessagesForRow;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    protected String getValidationErrorsForRow(List<String> validationErrors) {
+        String failureMessage = "Stale Check Row could not be processed for the following reasons:";
+        for (int i=0; i<validationErrors.size(); ++i) {
+            failureMessage += " [" + Integer.toString(i + 1) + "] " + validationErrors.get(i);
+        }
+        return failureMessage;
+    }
+
+    protected void validateCheckReconciliation(StaleCheckBatchRow staleCheckRow, CheckReconciliation checkReconciliation, List<String> validationErrors) {
+        if (!StringUtils.equalsIgnoreCase(staleCheckRow.getBankCode(), checkReconciliation.getBankCode())) {
+            validationErrors.add("The Bank Code in the file (" + staleCheckRow.getBankCode() + ") row does not match the bank on the Check (" +
+                    checkReconciliation.getBankCode() + ").");
+        }
+
+        try {
+            Double checkTotalAmountParsed = NumberFormat.getNumberInstance(java.util.Locale.US).parse(staleCheckRow.getCheckTotalAmount()).doubleValue();
+            KualiDecimal checkTotalAmount = new KualiDecimal(checkTotalAmountParsed);

```

Comment:
KualiDecimal has a constructor that accepts a String as input, so you don't need to parse the String value with NumberFormat explicitly.